src/net_processing.cpp

```diff
@@ -220,28 +223,65 @@ struct CNodeState {
     }
 };
 
+/** Provides locked access to a CNodeState */
+class CNodeStateAccessor {
+private:
+    std::shared_ptr<CNodeState> pstate;
+
+    CNodeStateAccessor() : pstate() { }
+    CNodeStateAccessor(std::shared_ptr<CNodeState>&& pstateIn) : pstate(std::move(pstateIn)) { ENTER_CRITICAL_SECTION(pstate->cs); }
```

Comment:
Hmm, looking again using CMutexLock means adding __FILE__, __LINE__, etc macros in net_processing for parameters to CMutexLock, instead of letting ENTER_CRITICAL_SECTION figure it out. If you prefer I could try to add some crazy defines like CONSTRUCT_CMUTEXLOCK(mutex, lockName) which fills in a constructor for use in the CNodeStateAccessor constructor?